File: SimpleLinearRegression.py
# For reading data set
# importing necessary libraries
import pandas as pd # deals with data frame  
import numpy as np  # deals with numerical values

wcat = pd.read_csv("E:\Data_Science_Okv\Datasets/wc-at.csv")

import matplotlib.pylab as plt #for different types of plots

plt.scatter(x=wcat['Waist'], y=wcat['AT'],color='green')# Scatter plot

np.corrcoef(wcat.Waist, wcat.AT) #correlation

help(np.corrcoef)

import statsmodels.formula.api as smf

model = smf.ols('AT ~ Waist', data=wcat).fit()
model.summary()


pred1 = model.predict(pd.DataFrame(wcat['Waist']))
pred1
print (model.conf_int(0.01)) # 99% confidence interval

res = wcat.AT - pred1
sqres = res*res
mse = np.mean(sqres)
rmse = np.sqrt(mse)
rmse

######### Model building on Transformed Data

# Log Transformation
# x = log(waist); y = at
plt.scatter(x=np.log(wcat['Waist']),y=wcat['AT'],color='brown')
np.corrcoef(np.log(wcat.Waist), wcat.AT) #correlation

model2 = smf.ols('AT ~ np.log(Waist)',data=wcat).fit()
model2.summary()

pred2 = model2.predict(pd.DataFrame(wcat['Waist']))
pred2
print(model2.conf_int(0.01)) # 99% confidence level

res2 = wcat.AT - pred2
sqres2 = res2*res2
mse2 = np.mean(sqres2)
rmse2 = np.sqrt(mse2)
rmse2
# Exponential transformation
plt.scatter(x=wcat['Waist'], y=np.log(wcat['AT']),color='orange')

np.corrcoef(wcat.Waist, np.log(wcat.AT)) #correlation

model3 = smf.ols('np.log(AT) ~ Waist',data=wcat).fit()
model3.summary()

pred_log = model3.predict(pd.DataFrame(wcat['Waist']))
pred_log
pred3 = np.exp(pred_log)
pred3
print(model3.conf_int(0.05)) # 99% confidence level

res3 = wcat.AT - pred3
sqres3 = res3*res3
mse3 = np.mean(sqres3)
rmse3 = np.sqrt(mse3)
rmse3

File: SimpleLinearRegression.R
wcat <- read.csv("E:/Data_Science_Okv/Datasets/wc-at.csv")
attach(wcat)

summary(wcat)

plot(Waist, AT) # scatter plot
cor(Waist, AT) # correlation coefficient
  
model <- lm(AT ~ Waist) # linear regression
model
summary(model) # output and evaluating

predict(model)
model$residuals

confint(model,level=0.99)
predict(model, interval = "confidence")

rmse <- sqrt(mean(model$residuals^2))
rmse


  # Log model
plot(log(Waist), AT)
cor(log(Waist), AT)
model2 <- lm(AT ~ log(Waist)) # log transformation
summary(model2)

rmse2 <- sqrt(mean(model2$residuals^2))
rmse2


# Exp model
plot(Waist, log(AT))
cor(Waist, log(AT))
model3 <- lm(log(AT) ~ Waist) # exponential tranformation
summary(model3)
model3$residuals

log_at <- predict(model3,interval="confidence")
log_at
at <- exp(log_at)
at

err <- AT-at
err

rmse3 <- sqrt(mean(err^2))
rmse3


# Polynomial transformation
model4 <- lm(log(AT) ~ Waist + I(Waist * Waist))
summary(model4)

confint(model4,level=0.95)

log_res <- predict(model4,interval="confidence")
atpoly <- exp(log_res)
atpoly
err_poly <- AT - atpoly
err_poly

rmse4 <- sqrt(mean(err_poly^2))
rmse4

